app/uk/gov/hmrc/uploaddocuments/controllers/BaseController.scala

```diff
@@ -86,4 +91,33 @@ abstract class BaseController(
       case _    => body
     }
 
+  final def withJourneyContext(
+    body: FileUploadContext => Future[Result]
+  )(implicit request: Request[_], ec: ExecutionContext): Future[Result] =
+    components.newJourneyCacheRepository.get(currentJourneyId)(DataKeys.journeyContext) flatMap {
+      case Some(journey) => body(journey)
+      case _             => Future.successful(Redirect(components.appConfig.govukStartUrl))
+    }
+
+  final def withUploadedFiles(
+    body: FileUploads => Future[Result]
+  )(implicit request: Request[_], ec: ExecutionContext): Future[Result] =
+    components.newJourneyCacheRepository.get(currentJourneyId)(DataKeys.uploadedFiles) flatMap {
+      case Some(files) => body(files)
+      case _           => Future.successful(Redirect(components.appConfig.govukStartUrl))
+    }
+
+  implicit class ResultExtensions(r: Result) {
+
+    //TODO: Not really sure what the flashing does - this was lifted from the Router class...
+    def withFormError(formWithErrors: Form[_]) = r.flashing(Flash {
+      val data = formWithErrors.data
+      // dummy parameter required if empty data
+      if (data.isEmpty) Map("dummy" -> "") else data
+    })
+  }
```

Comment:
Ohhh - I think this is used because we Redirect to another page - and want to show the form with Errors:
`Redirect(routes.ChooseMultipleFilesController.showChooseMultipleFiles).withFormError(formWithErrors)`

Reply:
I'm trying to get my head around how it works, but given that we're no longer using `formWithErrors.or(YesNoChoiceForm)` we might have broken this unhappy path scenario